source/portable/NetworkInterface/DriverSAM/gmac_SAM.c

```diff
@@ -624,29 +624,30 @@ uint32_t gmac_dev_read( gmac_device_t * p_gmac_dev,
         return GMAC_RX_NO_DATA;
     }
 
+    /* Return the number of bytes received. */
+    *p_rcv_size = bytesLeft;
```

Comment:
Same question in previous PR.

Is it necessary to compare with frame size?
The `*p_rcv_size` might be different when `ul_frame_size` is smaller than `bytesLeft + 2`.
```
if( bytesLeft + 2 <= ul_frame_size )
{
    *p_rcv_size = bytesLeft;
}
else
{
    *p_rcv_size = ul_frame_size;
}
```

Reply:
About:

~~~
    left = min( bytesLeft + 2, ( int32_t ) ul_frame_size );
~~~

In fact this code is never reached. It is kept in case someone wants to avoid zero-copy reception.

The following code works equally well, I tested it by disabling `ipconfigZERO_COPY_RX_DRIVER`:

~~~
    left = bytesLeft + 2;
~~~

When `gmac_dev_read()` is called, `ul_frame_size` equals `GMAC_RX_UNITSIZE`, which is 1536 bytes. The maximum RX size is 1514 bytes: no jumbo frames are allowed.

Your example should be:
~~~diff
    if( bytesLeft + 2 <= ul_frame_size )
    {
-        *p_rcv_size = bytesLeft;
+        *p_rcv_size = bytesLeft + 2;
    }
    else
    {
        *p_rcv_size = ul_frame_size;
    }
~~~

Reply:
Then it's good to me.
It'd be great if this info can be added to the code comments.